src/TransformOperationExecutor.ts

```diff
@@ -158,8 +158,9 @@ export class TransformOperationExecutor {
                 }
 
                 if (!this.isCircular(subValue, level)) {
+                    let transformKey = this.transformationType === "plainToClass" ? newValueKey : key;
```

Comment:
can you please explain this condition? Why do we need `newValueKey` only when `transformationType === "plainToClass"` ?